File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXTilePane.php
<?php
namespace php\gui\layout;

/**
 * @package php\gui\layout
 * @packages gui, javafx
 */
class UXTilePane extends UXPane
{
    /**
     * @var string HORIZONTAL or VERTICAL
     */
    public $orientation = 'HORIZONTAL';

    /**
     * @var float
     */
    public $hgap = 0.0;

    /**
     * @var float
     */
    public $vgap = 0.0;

    /**
     * @var float
     */
    public $prefTileWidth;

    /**
     * @var float
     */
    public $prefTileHeight;

    /**
     * @var int
     */
    public $prefColumns;

    /**
     * @var int
     */
    public $prefRows;

    /**
     * @var string
     */
    public $alignment = 'CENTER_LEFT';

    /**
     * @var string
     */
    public $tileAlignment = 'CENTER_LEFT';
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXTilePane.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.geometry.Orientation;
import javafx.geometry.Pos;
import javafx.scene.layout.TilePane;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "layout\\UXTilePane")
public class UXTilePane extends UXPane<TilePane> {
    interface WrappedInterface {
        @Property double hgap();
        @Property double vgap();

        @Property Pos alignment();
        @Property Pos tileAlignment();

        @Property double prefTileWidth();
        @Property double prefTileHeight();

        @Property int prefColumns();
        @Property int prefRows();

        @Property Orientation orientation();

    }

    public UXTilePane(Environment env, TilePane wrappedObject) {
        super(env, wrappedObject);
    }

    public UXTilePane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new TilePane();
    }
}
